src/main/java/com/codefactorygroup/betting/service/implementations/MarketServiceImpl.java

```diff
@@ -103,4 +103,34 @@ public MarketDTO updateMarket(final MarketDTO newMarket, final Integer marketId)
                 .map(MarketDTO::converter)
                 .orElseThrow(() -> new NoSuchEntityExistsException(String.format("Market with ID=%d doesn't exist.", marketId)));
     }
+
+    @Override
+    public List<Map<String, String>> calcAvgPriceForPreMatchMarkets() {
+        List<Market> markets = marketRepository.findAll();
+
```

Comment:
More functional way would be 

```
return marketRepository.findAll()
                .stream()
                .filter(x->checkThatMarketIsPrematch())
                .collect(Collectors.toMap(Market::getId, m -> m.getSelections()
                        .stream()
                        .mapToDouble(Selection::getOdds)
                        .average().getAsDouble()));
```

Reply:
@mihaidi Thank you. It's way better and cleaner